repose-aggregator/commons/utilities/src/test/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapperTest.scala

```diff
@@ -0,0 +1,1408 @@
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.MUTABLE, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.addHeader("a", "a")
+      wrappedResponse.addHeader("b", "b")
+
+      wrappedResponse.getHeaderNames should contain only("a", "b")
+    }
+
+    it("should not return any header names removed by succeeding interactions") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.MUTABLE, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.addHeader("a", "a")
+      wrappedResponse.addHeader("b", "b")
+      wrappedResponse.removeHeader("a")
+
+      wrappedResponse.getHeaderNames should contain only "b"
+    }
+
+    it("should return header names with the casing they were added with") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.MUTABLE, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.addHeader("A", "a")
+      wrappedResponse.addHeader("b", "b")
+
+      wrappedResponse.getHeaderNames should contain only("A", "b")
+    }
+  }
+
+  describe("getPreferredHeaders") {
+    it("should throw a QualityFormatException if the quality is not parseable") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.MUTABLE, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.addHeader("a", "a;q=fish")
+
+      a[QualityFormatException] should be thrownBy wrappedResponse.getPreferredHeaders("a")
+    }
+
+    it("should not return any headers added by preceding interactions") {
+      originalResponse.addHeader("a", "a")
+
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.PASSTHROUGH, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.getPreferredHeaders("a") should be('empty)
+    }
+
+    it("should return headers added by succeeding interactions") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.MUTABLE, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.addHeader("a", "a", 0.8)
+      wrappedResponse.addHeader("a", "b", 0.8)
+      wrappedResponse.addHeader("a", "c", 0.4)
+
+      wrappedResponse.getPreferredHeaders("a") should contain only("a", "b")
+    }
+
+    it("should not return any headers removed by succeeding interactions") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.MUTABLE, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.addHeader("a", "a")
+      wrappedResponse.removeHeader("a")
+
+      wrappedResponse.getPreferredHeaders("a") should be('empty)
+    }
+
+    it("should not return query parameters in the header value") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.MUTABLE, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.addHeader("A", "a;q=0.8;foo=bar")
+      wrappedResponse.addHeader("A", "b;bar=baz")
+
+      wrappedResponse.getPreferredSplittableHeaders("A") should contain only "b"
```

Comment:
It should absolutely be getPreferredHeaders, however it should still only contain "b" since it has a quality of 1.0 as opposed to "a"'s quality of 0.8.